Neos.Cache/Classes/Backend/MultiBackend.php

```diff
@@ -125,190 +110,155 @@ public function set(string $entryIdentifier, string $data, array $tags = [], int
     {
         $this->debug = $debug;
     }
 
     /**
-     * @param BackendInterface $backend
-     * @param string $entryIdentifier
-     * @param string $data
-     * @param array $tags
-     * @param int|null $lifetime
-     * @return void
-     * @throws \Neos\Cache\Exception
-     */
-    protected function setInBackend(BackendInterface $backend, string $entryIdentifier, string $data, array $tags = [], int $lifetime = null): void
-    {
-        $backend->set($entryIdentifier, $data, $tags, $lifetime);
-    }
-
-    /**
-     * @param BackendInterface $backend
-     * @param string $entryIdentifier
-     * @return mixed
+     * This setter is used by AbstractBackend::setProperties()
      */
-    protected function getFromBackend(BackendInterface $backend, string $entryIdentifier)
+    public function setRemoveUnhealthyBackends(bool $removeUnhealthyBackends): void
     {
-        return $backend->get($entryIdentifier);
+        $this->removeUnhealthyBackends = $removeUnhealthyBackends;
     }
 
     /**
-     * @param BackendInterface $backend
-     * @param string $entryIdentifier
-     * @return bool
+     * This setter is used by AbstractBackend::setProperties()
      */
-    protected function backendHas(BackendInterface $backend, string $entryIdentifier): bool
+    public function setLogErrors(bool $logErrors): void
     {
-        return $backend->has($entryIdentifier);
+        $this->logErrors = $logErrors;
     }
 
     /**
-     * @param BackendInterface $backend
-     * @param string $entryIdentifier
-     * @return bool
+     * @throws Throwable
      */
-    protected function removeFromBackend(BackendInterface $backend, string $entryIdentifier): bool
+    protected function handleError(Throwable $throwable): void
     {
-        return $backend->remove($entryIdentifier);
-    }
-
-    /**
-     * @param BackendInterface $backend
-     */
-    protected function flushBackend(BackendInterface $backend): void
-    {
-        $backend->flush();
+        if ($this->debug) {
+            throw $throwable;
+        }
     }
 
-    /**
-     * @param \Throwable $throwable
-     * @throws \Throwable
-     */
-    protected function handleError(\Throwable $throwable): void
+    protected function removeUnhealthyBackend(BackendInterface $unhealthyBackend): void
     {
-        if ($this->debug) {
-            throw $throwable;
+        if ($this->removeUnhealthyBackends === false || count($this->backends) <= 1) {
+            return;
+        }
+        foreach ($this->backends as $i => $backend) {
+            if ($backend === $unhealthyBackend) {
+                unset($this->backends[$i]);
+                $this->logger && $this->throwableStorage && $this->logger->warning(sprintf('Removing unhealthy cache backend %s from backends used by %s', get_class($unhealthyBackend), get_class($this)), LogEnvironment::fromMethodName(__METHOD__));
+                return;
+            }
         }
```

Comment:
Instead of a loop, `array_search(…)` could be used, it's even faster: https://3v4l.org/ubsXd

Reply:
Thanks, I fixed that.